elastic/kibana
# [Discover] Remove table column after that field was deleted

Closes https://github.com/elastic/kibana/issues/150958
Closes https://github.com/elastic/kibana/issues/151531

## Summary

This PR removes the field from table and sidebar after it was added as a column but deleted in the end.

Steps for testing:
- Create a runtime field
- Add to the table
- Delete the field and observe that it is removed from the table and sidebar.
